src/Microsoft.Health.Fhir.Api/Features/BackgroundJobService/JobFactory.cs

```diff
@@ -47,16 +52,36 @@ public JobFactory(IEnumerable<Func<IJob>> jobFactories, ILogger<JobFactory> logg
             }
         }
 
-        public IJob Create(JobInfo jobInfo)
+        public IScoped<IJob> Create(JobInfo jobInfo)
```

Comment:
JobFactory can now return the IJob instance and the Scope used to create IJob to JobHosting so it can be disposed when the run is complete. This gives an ambient scope to all classes that implement IJob.